eve/service/svc_note.py

```diff
@@ -0,0 +1,161 @@
+            self.note_data = json.load(f)
+            print(f'{self.note_file} loaded into class-object...')
+
+    def output(self):
+        # Create copy of note-data for output (because of the transformation-process)
+        output_data = self.note_data
+        # Transform None-values of output-data to str('None') and the other values to str()
+        for key in output_data:
+            for attribute in output_data[key]:
+                if output_data[key][attribute] is None:
+                    output_data[key][attribute] = 'None'
+                else:
+                    output_data[key][attribute] = str(output_data[key][attribute])
+        # Order output-data by 'Priority' and 'DueDate'
+        output_data = dict(sorted(output_data.items(), key=lambda x: x[1]['Priority']))
+        output_data = dict(sorted(output_data.items(), key=lambda x: x[1]['DueDate']))
+        # Create table for output
+        console = Console()
+        table = Table(show_header=True,
+                      box=box.ASCII,
+                      # show_lines=True,
+                      header_style='bold blue')
+        table.add_column('Key', style='dim')
+        table.add_column('CreationDate', style='dim')
+        table.add_column('EditDate', style='dim')
+        table.add_column('DueDate', style='dim')
+        table.add_column('Priority', style='dim')
+        table.add_column('NoteText')
+        # Populate table with output-data
+        for key in output_data:
+            row = list()
+            row.append(key)
+            row.append(output_data[key]['CreationDate'])
+            row.append(output_data[key]['EditDate'])
+            row.append(output_data[key]['DueDate'])
+            row.append(output_data[key]['Priority'])
+            row.append(output_data[key]['NoteText'])
+            table.add_row(str(row[0]), str(row[1]), str(row[2]), str(row[3]), str(row[4]), str(row[5]))
+        # Print table to console
+        console.print(table)
+
+    def add(self, note_text=None, due_date=None, priority=None):
+        # Compose new note
+        new_note = dict()
+        key = str(len(self.note_data) + 1)
+        new_note["CreationDate"] = str(date.today())
+        new_note["EditDate"] = None
+        new_note["DueDate"] = due_date
+        new_note["Priority"] = priority
+        new_note["NoteText"] = note_text
+        # Append new_note to note_data
+        self.note_data[key] = new_note
+        print(f'New note with key {key} added...')
+        # Save note_data to note_file
+        self.save()
+
+    def delete(self, key=None):
+        if key is None:
+            print(f'No key given...')
+        else:
+            # Delete key value from note_data for given key
+            try:
+                del self.note_data[key]
+                print(f'Key {key} deleted...')
+                # Assign new keys to note_data
+                new_keys = list()
+                for i in range(len(self.note_data)):
```

Comment:
Have you ever used the enumerate keyword, you can do...which will give you the index you're enumerating and the note. Might make this code a little simpler 

```suggestion
                for i, note in enumerate(self.note_data):
```

Reply:
Thanks for the hint. I didn't knew `enumerate`. Changed it.